cli/src/pcluster/models/imagebuilder.py

```diff
@@ -112,62 +161,78 @@ def __init__(self, image_name: str, config: dict = None, stack: ImageBuilderStac
         self.image_name = image_name
         self.__source_config = config
         self.__stack = stack
+        self.__image = None
         self.__config = None
         self.__bucket = None
         self.template_body = None
         self._s3_artifacts_dict = {
             "root_directory": "parallelcluster",
-            "root_image_directory": "imagebuilders",
-            "source_config_name": "image-config-original.yaml",
+            "root_image_directory": "images",
             "config_name": "image-config.yaml",
             "template_name": "aws-parallelcluster-imagebuilder.cfn.yaml",
             "custom_artifacts_name": "artifacts.zip",
         }
         self.__s3_artifact_dir = None
 
     @property
-    def s3_artifacts_dir(self):
+    def s3_artifact_dir(self):
         """Get s3 artifacts dir."""
         if self.__s3_artifact_dir is None:
-            self._get_artifact_dir()
+            self.__s3_artifact_dir = self._get_artifact_dir()
         return self.__s3_artifact_dir
 
+    @property
+    def config_url(self):
+        """Return configuration file S3 bucket url."""
+        return self.bucket.get_config_url(self._s3_artifacts_dict.get("config_name"))
+
     @property
     def stack(self):
         """Return the ImageBuilderStack object."""
         if not self.__stack:
             try:
-                self.__stack = ImageBuilderStack(AWSApi().instance().cfn.describe_stack(self.image_name))
-            except AWSClientError:
-                raise ImageBuilderActionError(f"ImageBuilder stack {self.image_name} does not exist.")
+                self.__stack = ImageBuilderStack(AWSApi.instance().cfn.describe_stack(self.image_name))
+            except AWSClientError as e:
+                if f"Stack with id {self.image_name} does not exist" in str(e):
```

Comment:
What about StackNotFoundError ? I like the approch you're using with ImageNotFoundError

Reply:
done, it's fine to raise StackNotFoundError, but need to remind that the difference is error message of describe_image is defined by ourselves, but the error message of describe stack is define by boto3 client. So if boto3 client change the error message, StackNotFoundError is still not reliable.